Sandbox sync tools

pbking opened the pull request:
Now that the GIT and SVN repositories are in sync I'm working on some tooling and workflows to ensure that they stay that way.

#### Changes proposed in this Pull Request:

This change adds a 'sandbox.sh' script that helps to move LOCAL changes to SVN both for DEVELOPMENT or to DEPLOY CODE in preparation for production.

USAGE of the script is outlined in updates to README.md

pbking committed:
First commit of sync tools.

Not ready for prime time (still have user-specific info in the script)
Two scripts, one to clean and update the sandbox,
another to sync the project to the sandbox ignoring files in the ignore
file.

pbking committed:
Improved sandbox scripts

Moved all sandbox script logic to a singular sandbox.sh script
Moved sandbox configuration information to an ignored .sandbox-config
Updated README with sandbox script usage instructions
Added a sandbox:watch option to package.json
added sync/clean options to package.json scripts

MaggieCabrera commented:
When I run the command for the first time I get this at the end:

```
Writing .sandbox-config
sandbox.sh: line 33: [: ==: unary operator expected
sandbox.sh: line 41: [: ==: unary operator expected
No known command given. [clean, push]
```

When I run clean I get:

```
sh sandbox.sh clean
Pseudo-terminal will not be allocated because stdin is not a terminal.
```

Still, both commands seem to be working ok. The config file is created, the clean command seems to be working fine.

MaggieCabrera commented:
This is not related to this commit per se, but with the GH sync. When I use the push command this is what gets pushed to SVN:
```
M       README.md
M       apostrophe-2/index.php
M       brompton/package-lock.json
M       brompton/package.json
M       brompton/sass/_extra-child-theme.scss
M       brompton/sass/style-child-theme.scss
M       brompton/style-rtl.css
M       brompton/style.css
M       package-lock.json
M       package.json
?       sandbox.sh
?       seedlet/assets/sass/blocks/blog-posts
?       spearhead/spearhead.zip
```

`apostrophe-2/index.php` is the only file I made a change to test the push. I'm doing this from this branch after rebasing trunk

MaggieCabrera commented:
After all my testing I did a clean again and I get a long list of files being reverted from our previous sync.

Here is such list: 27281-pb/

pbking committed:
Adding a trunk up-to-date warning to push

pbking committed:
Improved sync and clean logic.

Determined files to IGNORE by comparing what changed between the current
BRANCH divergence point and the TIP of TRUNK

Cleaned up clean logic output

pbking committed:
Added options to push option

pbking committed:
Improved how files are ignored when pushing

pbking committed:
Improved README

pbking committed:
Reformat readme

pbking commented:
@MaggieCabrera I believe that the issues you raised are addressed if you have an opportunity to try that again.

scruffian reviewed (approved):
This looks great. 🚢

jffng commented on .sandbox-ignore:
```diff
@@ -0,0 +1,8 @@
+.sandbox-ignore
+sandbox.sh
+vendor
+node_modules
```
Could we add `*.zip` to this ignore?

```suggestion
node_modules
*.zip
```

Also on my first time sync'ing, this repo's README.md is pushed up.

jffng reviewed (approved):
This is so helpful! 

In the future it may be useful to extract this into an npm package or standalone repo so that it can be incorporated in other directories, but in the meantime this LGTM 🚢
